### DISCUSSION TITLE
How to disable PWM after invoked?

### DISCUSSION THREAD

**Discussion by lexcibien** [earlephilhower/arduino-pico#2350] [Q&A] [answered]
I invoked the PWM on this line`if(PWM_isEnabled) {generatePWMSignal(PWM_frequency, PWM_dutyCycle);}` it's working normally, except when i get the frequency to arround 2000Hz, then it's stoped, but i manage to limit to 2000Hz since i didn't needed beyond that. My function is that one down.

`void generatePWMSignal(uint16_t frequency, uint8_t dutyCycle) {
  uint32_t duty = ((uint32_t)(dutyCycle) * (FREQ_BITS_RANGE - 3)) / 100 + 3;

  analogWriteFreq(frequency);
  analogWrite(PWM_PIN, duty);
}`
When i call `generatePWMSignal(frequency, dutyCycle);` it runs, but i need to stop after running the signal, how can i do that?

**Comment by earlephilhower** [comment, marked as answer]
Just do a`digitalWrite` on the pin to disable PWM.

**Reply by lexcibien** [reply]
Well, thanks didn't think on that!
